src/Model/Parser.cs

```diff
@@ -1,28 +1,28 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
-using System.Linq;
+using System.Text.RegularExpressions;
 using FuncSharp;
 
 namespace Mews.Fiscalization.SignatureChecker.Model
 {
     internal static class Parser
     {
         private static readonly CultureInfo FrenchCulture = new CultureInfo("fr-FR");
+        private static readonly Regex AmountRegex = new Regex(@"([-−]?[\d\s\\.,]*[\d\\.,])\s*([^\d\s\.]+)$");
 
         public static ITry<decimal, IEnumerable<string>> ParseDecimal(string value)
         {
-            var isSuccess = Decimal.TryParse(value.Replace('.', ',').Trim(), NumberStyles.Number, FrenchCulture, out var result);
+            var isSuccess = Decimal.TryParse(value.Replace('.', ',').Replace('−', '-').Trim(), NumberStyles.Number, FrenchCulture, out var result);
             return isSuccess.ToTry(_ => result, _ => "Invalid number.".ToEnumerable());
         }
 
         public static ITry<Amount, IEnumerable<string>> ParseAmount(string stringValue)
         {
-            var tokens = stringValue.Split('\u00A0', ' ').Where(t => !String.IsNullOrWhiteSpace(t)).ToList();
-            return tokens.Count.Match(
-                2, _ => ParseDecimal(tokens[0]).FlatMap(value => Amount.Create(value: value, currencyCodeOrSymbol: tokens[1].Trim())),
-                _ => Try.Error<Amount, IEnumerable<string>>($"Invalid {nameof(Amount)}.".ToEnumerable())
-            );
+            var amountParts = AmountRegex.Match(stringValue);
+            var value = amountParts.Groups[1].Value;
+            var currencyCodeOrSymbol = amountParts.Groups[2].Value;
```

Comment:
Doesn't this actually throw exception when it doesn't match?
Can we at least wrap it in trycatch?

Reply:
It will actually give you an empty string, which we can't parse. I improved the regex so it we always get a correct error message.